vortex-scan/src/work_queue.rs

```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
+    #[test]
+    fn fuzz_edge_cases() {
+        for _ in 0..50 {
+            let mut rng = rand::rng();
+            let scenario = rng.random_range(0..6);
+
+            match scenario {
+                0 => test_all_empty_factories(rng.random_range(1..=10), rng.random_range(1..=4)),
+                1 => {
+                    test_single_task_per_factory(rng.random_range(1..=20), rng.random_range(1..=8))
+                }
+                2 => test_single_factory_many_workers(rng.random_range(2..=8)),
+                3 => test_many_factories_single_worker(rng.random_range(5..=50)),
+                4 => test_interleaved_factory_loading(
+                    rng.random_range(5..=15),
+                    rng.random_range(2..=4),
+                ),
+                _ => test_rapid_worker_creation(rng.random_range(10..=20)),
+            }
+        }
+    }
+
+    // Helper function to create a factory that produces a specific number of tasks
+    fn create_factory(
+        task_count: usize,
+        id: usize,
+        delay_ms: u64,
+        should_fail: bool,
+    ) -> TaskFactory<(usize, usize)> {
+        Box::new(move || {
+            if delay_ms > 0 {
+                thread::sleep(Duration::from_millis(delay_ms));
+            }
+
+            if should_fail {
+                vortex_bail!("Factory failed");
+            }
+
+            Ok((0..task_count).map(|i| (id, i)).collect())
+        })
+    }
+
+    fn test_concurrent_processing(
+        num_factories: usize,
+        num_workers: usize,
+        tasks_per_factory: usize,
+    ) {
+        let mut factories = Vec::new();
+        for i in 0..num_factories {
+            factories.push(create_factory(tasks_per_factory, i, 0, false));
+        }
+
+        let queue = WorkQueue::new(factories);
+        let expected_total = num_factories * tasks_per_factory;
+
+        // Spawn workers
+        let barrier = Arc::new(Barrier::new(num_workers));
+        let processed_count = Arc::new(AtomicUsize::new(0));
+        let task_tracking = Arc::new(std::sync::Mutex::new(
+            HashMap::<(usize, usize), usize>::new(),
+        ));
+
+        let handles: Vec<_> = (0..num_workers)
+            .map(|_worker_id| {
+                let queue = queue.clone();
+                let barrier = barrier.clone();
+                let processed_count = processed_count.clone();
+                let task_tracking = task_tracking.clone();
+
+                thread::spawn(move || {
+                    barrier.wait();
+                    let mut iterator = queue.new_iterator();
+                    let mut local_count = 0;
+
+                    while let Some(task_result) = iterator.next() {
+                        match task_result {
+                            Ok(task) => {
+                                // Track this task to detect duplicates
+                                {
+                                    let mut tracking = task_tracking.lock().unwrap();
```

Comment:
As we don't care about perf in the test, let's drop the extra scope for readability.